Makefile.base

```diff
@@ -63,6 +63,12 @@ ifeq ($(strip $(ASSMSRC))$(NO_AUTO_SRC),)
   ASSMSRC := $(wildcard *.S)
 endif
 
+ifneq (,$(SRCXX))
+  ifeq (,$(filter cpp,$(FEATURES_USED)))
+    $(error Found C++ source, but feature "cpp" is not used. Add "FEATURES_REQUIRED += cpp")
```

Comment:
Instead of outputting a warning here, can't we simply do

```Makefile
ifneq (,$(SRCXX))
  FEATURES_REQUIRED += cpp
endif
```

here?

Reply:
That would it make more of a bug fix :-).

Reply:
Dependency resolution is not supposed to happen at that level. So I would say no.

Reply:
Will also not work: The source detection is done after the dependency resolution module by module

Reply:
(I would have been in favor of this, otherwise. But restructuring the build system seems to be too much of work for this.)